### DISCUSSION TITLE
Tycho 2.7.2 / 2.7.3 / 2.7.4 / 2 .7.5 bugfix release

### DISCUSSION THREAD

**Discussion by akurtakov** [eclipse-tycho/tycho#903] [Announcements]
We are preparing a bugfix release of Tycho 2.7.2 to include some important fixes.

For details of included bugfixes, see the [release notes](https://github.com/eclipse/tycho/blob/tycho-2.7.x/RELEASE_NOTES.md), please let us know if you are aware of other regressions you would like to have included.

Also please help by testing the SNAPSHOTS build. To use it, change your Tycho version to 2.7.4-SNAPSHOT and add the following snippet to your pom.

```
<pluginRepositories>
    <pluginRepository>
      <id>tycho-snapshots</id>
      <url>https://repo.eclipse.org/content/repositories/tycho-snapshots/</url>
    </pluginRepository>
</pluginRepositories>
```

You can find the current planned release date and progress [here](https://github.com/eclipse/tycho/milestone/7).

Regards,
Tycho team

**Comment by HannesWell** [comment]
M2E build looks good with the current Tycho 2.7.2-SNAPSHOT:
https://github.com/eclipse-m2e/m2e-core/pull/657

**Reply by HannesWell** [reply]
I also tested the new `Support for PDE Declarative Component Annotation progressing` and removed the component's xml under OSGI-INF as well as the import of package `org.osgi.service.component.annotations` and everything worked fine. 👍🏽

**Comment by laeubi** [comment]
### Tycho 2.7.2 is released

Tycho 2.7.2 has been released and is available from Maven Central repository.

🆕 https://github.com/eclipse/tycho/blob/tycho-2.7.2/RELEASE_NOTES.md
🏷️ https://github.com/eclipse/tycho/tree/tycho-2.7.2
👔 https://projects.eclipse.org/projects/technology.tycho/releases/2.7.2
🙏 contributors who contributed patches for this release:

- 	Christoph Läubrich
- 	Hannes Wellmann

and thanks and to everyone who helped us with testing the staged version.

Regards,
The Tycho Team

**Comment by laeubi** [comment]
I'm preparing a 2.7.3 because of 
- https://github.com/eclipse/tycho/issues/932

announcement will be made soon... as this does only include local changes that are mostly independent and covered by integration-tests I plan to release this on monday without special testing period.

**Comment by laeubi** [comment]
### Tycho 2.7.3 is released

Tycho 2.7.3 has been released and is available from Maven Central repository.

🆕 https://github.com/eclipse/tycho/blob/tycho-2.7.3/RELEASE_NOTES.md
🏷️ https://github.com/eclipse/tycho/tree/tycho-2.7.3
👔 https://projects.eclipse.org/projects/technology.tycho/releases/2.7.3
🙏 contributors who contributed patches for this release:

-    Christoph Läubrich

and thanks and to everyone who helped us with testing the staged version.

Regards,
The Tycho team

**Comment by laeubi** [comment]
I plan to release a bugfix release of Tycho 2.7.4 most probably at end of next week, snapshots are available shortly please let me know about any issues or relevant fixes.

**Reply by cdietrich** [reply]
I can’t find a 2.7.4 snapshot 

Error:      Unresolveable build extension: Plugin org.eclipse.tycho:tycho-maven-plugin:2.7.4-SNAPSHOT or one of its dependencies could not be resolved: Could not find artifact org.eclipse.tycho:tycho-maven-plugin:jar:2.7.4-SNAPSHOT in tycho-snapshots (https://repo.eclipse.org/content/repositories/tycho-snapshots/) -> [Help 2]

**Reply by laeubi** [reply]
Tanks for the feedback, the deploy build is not running as expected, I'll take a look...

**Reply by laeubi** [reply]
Snapshots should be deployed now!

**Reply by cdietrich** [reply]
thx.can continue evaluation now

**Reply by HannesWell** [reply]
https://github.com/eclipse/tycho/pull/1020 would be nice to have. It should be simply cherry-pickable to the 2.7.x branch.

**Reply by laeubi** [reply]
@HannesWell do you like to open a PR against 2.7.x branch?

**Comment by laeubi** [comment]
I'll postpone the release by one week as we found a bug in tycho-pomless (but not a regression), that should be included.

**Comment by laeubi** [comment]
Tycho 2.7.4 has been released and is available from Maven Central repository.

🆕 https://github.com/eclipse/tycho/blob/tycho-2.7.4/RELEASE_NOTES.md
🏷️ https://github.com/eclipse/tycho/tree/tycho-2.7.4
👔 https://projects.eclipse.org/projects/technology.tycho/releases/2.7.4
🙏 contributors who contributed patches for this release:

- 	Alexander Kurtakov
- 	Christoph Läubrich
- 	Hannes Wellmann

and thanks and to everyone who helped us with testing the staged version.

Regards,
the Tycho team

**Reply by HannesWell** [reply]
Thank you @laeubi for running the release!
Not sure if this is relevant, but I noticed the release date is from last week (probably the date planned first).

Furthermore I noticed that the release tags on the 2.7.x branch are not on commits that are in the branch.
Wouldn't it be better to include the release commits in the branch. This would either have the consequence that the branch would include the release commit (where all -SNAPSHOT suffixes are removed from the version) and the prepare for next development iteration commits or that not the real commits from which the build is done are tagged.  But the former would IMHO OK since the versions have to be bumped anyways. At the moment one has to fetch the tags separately (which is from my experience not done by the usual git config) and it gives a not so nice history with even more branches.

**Reply by laeubi** [reply]
> I noticed that the release tags on the 2.7.x branch are not on commits that are in the branch.

Whats the advantage of those? If one really want he can checkout the tag and branch from that, but usually we simply spin of a new release instead of doing a 2.7.3.SR1 (or what ever)... so we should really not complicate this more as necessary...

**Reply by HannesWell** [reply]
> > I noticed that the release tags on the 2.7.x branch are not on commits that are in the branch.
> 
> Whats the advantage of those? If one really want he can checkout the tag and branch from that, but usually we simply spin of a new release instead of doing a 2.7.3.SR1 (or what ever)... so we should really not complicate this more as necessary...

It is sometimes helpful if one wants to debug a certain version.
But I don't know the release process exactly and it is not important, so yes, no need to make it more complicated. I just wanted to mention it.

**Reply by laeubi** [reply]
> It is sometimes helpful if one wants to debug a certain version.

in that case one can simply checkout the TAG, or even the commit before the change to the next development version.

> But I don't know the release process exactly

See https://github.com/eclipse/tycho/blob/master/devdoc/Releasing_Tycho.md but its not put in stone ... it just has worked in the past (keep in mind that we actually never have had bugfix-releases before).

**Reply by HannesWell** [reply]
> > It is sometimes helpful if one wants to debug a certain version.
> 
> in that case one can simply checkout the TAG, or even the commit before the change to the next development version.

That latter was what I did, in the first moment I did not think of fetching the tags, so it was a bit inconvenient, but definitely nothing serious.

> 
> See https://github.com/eclipse/tycho/blob/master/devdoc/Releasing_Tycho.md but its not put in stone ... it just has worked in the past (keep in mind that we actually never have had bugfix-releases before).

Maybe after the `git add * && git commit version change` push that change, respectively run the preparation for the next release-cycle on top of that change and push it then? Btw. you are probably aware of the `maven-release-plugin`, which creates those two commits? I know we had a discussion in the past that it does not work well with Eclipse Plug-in projects and Tycho, but does the same restriction apply for Tycho itself?
However this is nothing urging or something with high demand, so no need to invest time into that. :)

**Comment by HannesWell** [comment]
I just noticed that the `pack200` plug-ins were removed from Tycho 2.7.3 to 2.7.4. Was this intentional? I only found a note about the removal for 3.0.0.
See https://repo1.maven.org/maven2/org/eclipse/tycho/extras/tycho-pack200/

**Reply by laeubi** [reply]
If one feels it missing, it should be possible to deploy just that plugins.... But I have no clue why I did the deploy as usual...

**Reply by HannesWell** [reply]
I don't need it directly. I just updated the Eclipse-Passage build to Tycho 2.7.4 to be able to remove the Tycho-connector. But they had to revert that because they still use pack200. But it looks like they want to remove that support any way:
https://github.com/eclipse-passage/passage/pull/1114

So personally I don't have a strong need for that. But as Aleks suggested, if we accept that it is gone now, at least it should be documented.

**Reply by laeubi** [reply]
You can use an older Tycho release for the pack200 as well :-)

**Comment by akurtakov** [comment]
Hmm, that was not the plan and I have no idea how/why this happened. But now that it happened it might be best to just document it as pack200 is done deal and even p2 no longer supports it.

**Comment by laeubi** [comment]
I have setup the 2.7.x branch or a 2.7.5 bugfix release, so please open PRs for anything that should be fixed/backported.

@HannesWell do you like to take a look if you can see why pack200 is missing?

**Reply by HannesWell** [reply]
It is probably missing because the `tycho-extras` pom only includes the `pack200` if the JDK version (running the build?) is below 13:
```
	<profiles>
		<profile>
    		<id>pack200</id>
    		<activation>
        		<jdk>(,13]</jdk>
    		</activation>
			<modules>
				<module>pack200</module>
			</modules>
		</profile>
	</profiles>
```

So the removal is probably because the release build was run with JDK>13. Running it with JDK-11 should re-include it.

But personally I don't have a need for pack200 support anymore.

**Reply by HannesWell** [reply]
Just quickly checked that pack200 is back again for 2.7.5:
https://repo1.maven.org/maven2/org/eclipse/tycho/extras/tycho-pack200/2.7.5/

Thank you Christoph.

**Reply by laeubi** [reply]
I enforced using a Java 11 VM this time, so that seems to be the issue here!

**Comment by globavi** [comment]
Hi, 
is there already a rough idea when 3.0.0 will be officially relased? (if rather beginning or end of Q4/ next year)?

**Reply by akurtakov** [reply]
Goal is to release by end of this month (Sep 2022).

**Reply by globavi** [reply]
thanks for quick reply: )

**Reply by laeubi** [reply]
See https://github.com/eclipse-tycho/tycho/milestone/5

**Comment by laeubi** [comment]
Tycho 2.7.5 has been released and is available from Maven Central repository.

🆕 https://github.com/eclipse-tycho/tycho/blob/tycho-2.7.5/RELEASE_NOTES.md
🏷️ https://github.com/eclipse-tycho/tycho/tree/tycho-2.7.5
👔 https://projects.eclipse.org/projects/technology.tycho/releases/2.7.5
🙏 contributors who contributed patches for this release:

-  Christoph Läubrich
-  Mickael Istria

and thanks and to everyone who helped us with testing the staged version.

Regards,
 The Tycho Team
